apps/frontend-pwa/src/components/learningElements/QuestionStack.tsx

```diff
@@ -143,16 +146,58 @@ function QuestionStack({
           [element.id]: '',
         }))
       }
+    })
+  }, [stack])
 
-      if (element.questionInstance?.evaluation) {
-        setIsEvaluation(true)
+  // check if the current stack is showing evaluation data and which elements were answered correctly
+  useEffect(() => {
+    if (
+      stack.elements?.every(
+        (element) =>
+          !element.questionInstance || element.questionInstance.evaluation
+      ) ??
+      false
+    ) {
+      setIsEvaluation(true)
+    }
+
+    let overallScore: 'correct' | 'incorrect' | 'partial' | 'unanswered' =
+      'unanswered'
+    stack.elements?.forEach((element) => {
```

Comment:
maybe generalize this logic and use reduce

Reply:
the entire logic has been reworked into a single and more concise useEffect snipped with reduced complexity and better readability